### ISSUE TITLE
how to duplicate or drop frames when using setpts filter?

### ISSUE THREAD

**Issue by allenling** [PyAV-Org/PyAV#1247]
## Overview

when we used setpts filter to slow down a video, we found that setpts would not help duplicate frames itself.

```
def test_raw_setpts(vid_path ):
    incontainer = av.open(vid_path, "r")
    g = Graph()
    params = {"expr": f"2.0*PTS"}
    setpts = g.add("setpts", **params)
    buf = g.add_buffer(template=incontainer.streams.video[0])
    buf.link_to(setpts, 0, 0)
    sink = g.add('buffersink')
    setpts.link_to(sink, 0, 0)
    g.configure()
    #
    c = 0
    for frame in incontainer.decode(video=0):
        c += 1
    print("count", c)
    return
```

the total number of frames from setpts was expected to be double, in above case, the count should be double as frames in input file.

but acutally the count was unchanged.

so we need to drop or duplicate frames on our own?

**Comment by WyattBlue** [comment]
![image](https://github.com/PyAV-Org/PyAV/assets/57511737/676cfbd1-b880-4ea1-a9a8-348e29ee9559)

**Comment by allenling** [comment]
@WyattBlue 

em.... we found out that setpts is just meant to , as its name suggests, change pts. more precisely, what it does is to multiply 

the original pts by a float factor, and that's all.

duplicating and dropping frames are done by ffmpeg in some functions, so far, we are not clear how ffmpeg handle dropping

and duplicating. so this is not a problem of pyav, that's why we closed this issue.

if there's any advice or tips on without changing fps, duplicating and dropping frames when speeding up and slowing 

down videos to share with us, we really appreciate that.
